### DISCUSSION TITLE
Command line arguments to make a PDF

### DISCUSSION THREAD

**Discussion by dougransom** [ImageMagick/ImageMagick#5982] [Help]
### ImageMagick version

7.1.0-57

### Operating system, version and so on

windows

### Description

I have posted this in freelancer  https://www.freelancer.com/projects/shell-script/Image-Magick-command-line-arguments/details.  

I have an an image, could be jpeg, png, bitmap etc, on windows. Varying width and height and resolution.

I will run the solution in powershell.

I would like the image placed onto a PDF 8.5"x11", so the width of the image is scaled 6" wide, and the right and top margin are 1" and 1". A command line sample assuming the name "a.png" for the input image will be sufficient, or a powershell script. I would like to specify the output pdf file name as well on the command line.

I would also like a command line sample to rotate the image -90, 0, 90, or 180 degrees whether it is image magick or some other powershell command.

**Comment by snibgo** [comment]
The fundamental units of IM is "pixels", not "inches". The margins of the image will be white or transparent pixels. You need to decide on how many pixels are required for your 8.5"x11" image, and on how this will be calculated. For example, you might decide the PDF is to be 300dpi, so the width in pixels will be 8.5*300=2550, and your input image will need to be resized up or down.

Or you might decide you don't want to resize your input, and don't care what the dpi is, so you want to add margins to make the output aspect ratio 8.5:11, so will calculate dpi from the size in pixels and required inches.

Or you might want something different.
